addons/hr_timesheet/report/report_timesheet_templates.xml

```diff
@@ -85,11 +85,13 @@
                     <t t-foreach="docs" t-as="doc">
                         <div class="oe_structure"/>
                         <div class="mt8">
-                            <h1>
-                                <t t-out="title"/>: <t t-out="doc.name"/>
-                            </h1>
                             <t t-set="lines" t-value="doc.timesheet_ids if from_project else timesheets_per_task.get(doc, doc.env['account.analytic.line'])"/>
-                            <t t-call="hr_timesheet.timesheet_table"/>
+                            <t  t-if="lines">
+                                <h1>
+                                    <t t-out="title"/>: <t t-out="doc.name"/>
+                                </h1>
+                                <t t-call="hr_timesheet.timesheet_table"/>
+                            </t>
```

Comment:
this code has a small risk of breaking XPATH (since it change the structure a little) but it's better than removing nodes so should be rather safe

this could be written more safely with eg:

```suggestion
                            <h1 t-if="lines">
                                <t t-out="title"/>: <t t-out="doc.name"/>
                            </h1>
                            <t  t-if="lines" t-call="hr_timesheet.timesheet_table"/>
```
that would keep the same HTML structure, just move the t-set above (which should not break any sane XPATH)